aptos-move/framework/aptos-framework/sources/native_bridge.move

```diff
@@ -141,28 +162,34 @@ module aptos_framework::native_bridge {
     /// @param amount The amount of assets to be locked.  
     /// @param nonce The unique nonce for the transfer.    
     /// @abort If the caller is not the bridge relayer or the transfer has already been processed.  
-    public entry fun complete_bridge_transfer(  
-        caller: &signer,  
+    public entry fun complete_bridge_transfer(
+        caller: &signer,
         bridge_transfer_id: vector<u8>,
-        initiator: vector<u8>,  
-        recipient: address,  
-        amount: u64,  
-        nonce: u64  
-    ) acquires BridgeEvents {  
+        initiator: vector<u8>,
+        recipient: address,
+        amount: u64,
+        nonce: u64
+    ) acquires BridgeEvents {
         // Ensure the caller is the bridge relayer
-        native_bridge_configuration::assert_is_caller_relayer(caller);  
+        native_bridge_configuration::assert_is_caller_relayer(caller);
 
         // Check if the bridge transfer ID is already associated with an inbound nonce
         let inbound_nonce_exists = native_bridge_store::is_inbound_nonce_set(bridge_transfer_id);
-        assert!(!inbound_nonce_exists, ETRANSFER_ALREADY_PROCESSED); // Abort if the transfer is already processed
-        assert!(nonce > 0, EINVALID_NONCE); 
+        assert!(!inbound_nonce_exists, ETRANSFER_ALREADY_PROCESSED);
+        assert!(nonce > 0, EINVALID_NONCE);
 
         // Validate the bridge_transfer_id by reconstructing the hash
+        let initiator_bytes = native_bridge_store::hex_to_bytes(initiator);
+        let recipient_bytes = bcs::to_bytes(&recipient);
+        let amount_bytes = native_bridge_store::normalize_to_32_bytes(bcs::to_bytes<u64>(&amount));
+        let nonce_bytes = native_bridge_store::normalize_to_32_bytes(bcs::to_bytes<u64>(&nonce));
+
         let combined_bytes = vector::empty<u8>();
-        vector::append(&mut combined_bytes, bcs::to_bytes(&initiator));
-        vector::append(&mut combined_bytes, bcs::to_bytes(&recipient));
-        vector::append(&mut combined_bytes, bcs::to_bytes(&amount));
-        vector::append(&mut combined_bytes, bcs::to_bytes(&nonce));
+        vector::append(&mut combined_bytes, initiator_bytes);
```

Comment:
this encoding is repeated a few times, could be a view method in native_bridge

Reply:
It looks repetitive, but the data types are different in each case, like sometimes recipient is a Move address and sometimes it's an Eth address... so it would need at least two helper functions to refactor it. Unless you think it's essential to refactor I'd leave it as it is for now.